### PR TITLE
fix fillto for barplot

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

File: src/basic_recipes/basic_recipes.jl [hunk 1]
```diff
@@ -505,20 +505,28 @@ end
 
 conversion_trait(::Type{<: BarPlot}) = PointBased()
 
+function bar_rectange(xy, width, fillto)
```
Comment (asinghvi17) [inline src/basic_recipes/basic_recipes.jl]:
```suggestion
function bar_rectangle(xy, width, fillto)

File: src/basic_recipes/basic_recipes.jl [hunk 2]
```diff
@@ -505,20 +505,28 @@ end
...
-            hw = mean(diff(first.(xy))) # TODO ignore nan?
+        if width === automatic
+            # times 0.8 for default gap
+            width = mean(diff(first.(xy))) * 0.8 # TODO ignore nan?
         end
 
-        return FRect.(first.(xy) .+ (hw ./ -2f0),
-                      Float32.(fillto), abs.(hw),
-                      last.(xy) .- Float32.(fillto))
+        return bar_rectange.(xy, width, fillto)
```
Comment (asinghvi17) [inline src/basic_recipes/basic_recipes.jl]:
```suggestion
        return bar_rectangle.(xy, width, fillto)

**Comment by SimonDanisch** [comment]
The barplot image changed, since I added gaps between them (@mkborregaard forced me to :P ):
image

**Comment by mkborregaard** [comment]
💖

**Pull request merged**